server/src/main/java/com/skillmagnet/User/UserController.java

```diff
@@ -32,4 +55,47 @@ public ResponseEntity<User> getUser(@RequestParam("username") String username) {
         return new ResponseEntity<>(user, HttpStatus.OK);
     }
 
+    /**
+     * Retrieve a user by id.
+     * @param id - user's unique int id
+     * @return
+     *          Success: Requested User
+     *          Failure: NOT_FOUND
+     */
+    @GetMapping("/user/{id}")
+    public ResponseEntity<User> getUser(@PathVariable("id") int id) {
+        User user = userRepository.findById(id);
+
+        if (user == null) {
+            return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+        }
+
+        return new ResponseEntity<>(user, HttpStatus.OK);
+    }
+
+    /**
+     * Compares login credentials to stored credentials.
+     * @param username - name of user attempting login
+     * @param password - password provided at login (Plaintext)
+     * @return
+     *         Success(Match): "Passwords Match!"
+     *         Success(Mismatch): "Passwords do not match!"
+     *         Failure: NOT_FOUND
+     */
+    @PutMapping("/user/login")
+    public ResponseEntity<String> loginUser(@RequestParam("username") String username,
+                                              @RequestParam("password") String password) {
```

Comment:
:memo: Refer to above. Gonna want this to be a `POST` request and use a `@RequestBody User` and grab the username / password details from that object. 

`POST` requests allow you to send request bodies and is generally the standard method for logging in a user. 

❗Additionally, you want to send passwords in the request body so the data is encrypted. As a request param you'd be doing something `PUT skillmagnet/api/user/login?username=bob&password=password123` which is just sent out onto the network and anyone can see those personal details. Sending info as request bodies will encrypt the payload.

Reply:
Okay yeah that makes a lot more sense! Appreciate the info!